actionpack/lib/action_controller/metal/renderers.rb

```diff
@@ -18,16 +18,33 @@ def initialize(format)
     end
   end
 
+  class MissingSerializingFormat < KeyError
+    def initialize(format)
+      super "No serializing format defined for: '#{format}'."
+    end
+  end
+
   module Renderers
     extend ActiveSupport::Concern
 
     # A Set containing renderer names that correspond to available renderer procs.
     # Default values are <tt>:json</tt>, <tt>:js</tt>, <tt>:xml</tt>.
     RENDERERS = Set.new
+    FORMATTERS = Hash.new do |h, format|
+        fail ActionController::MissingSerializingFormat.new(format)
+    end
 
     included do
       class_attribute :_renderers
       self._renderers = Set.new.freeze
+      class_attribute :_formatters
+      self._formatters = FORMATTERS.dup.freeze
+    end
+
+    class_methods do
+      def serializing(**formatters)
+        self._formatters = _formatters.merge(formatters)
```

Comment:
I see `remove_formatter` only changes the default value. I think we are fine without one.